### DISCUSSION TITLE
Add a way to indicate that a type/document has been generated

### DISCUSSION THREAD

**Discussion by baywet** [typescript-eslint/typescript-eslint#7589] [Community Feedback]
Hi everyone 👋

I'm working on [kiota](https://github.com/microsoft/kiota) an OpenAPI based and open-source client code generator for REST APIs.
As we're generating code for users, we don't want that code to trigger linting rules. We of course strive to produce the cleanest code as possible, but even then, linting configuration from one project to another might differ.

[CSharp has the auto-generated doc comment](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-migration-strategies) and [Java has the generated annotation](https://tomcat.apache.org/tomcat-10.0-doc/annotationapi/jakarta/annotation/Generated.html) both effectively disable any warn/error/info behavior from the target files/types allowing the developer to focus on their code.

We prefer that approach over project configuration as it reduces setup instructions and gets people to calling their API faster.

With that context in mind: is there an equivalent with eslint/ts-parser? If so, would you mind pointing me to it (couldn't find it during my searches), if not, is this something that should be considered by the broader community here? (or should I post that in eslint directly)

Thanks a lot!

**Comment by bbarry** [comment]
I like to wrap generated files with:

```js
/* eslint-disable */
/* generated by ... */

...

/* eslint-enable */
```

**Reply by baywet** [reply]
thanks a lot! that's probably all we need at this point!

**Comment by bradzacher** [comment]
Bill has answered your question - though for future reference - as per our contributing guide we don't use discussions as a support forum. Please see our contributing guide for better alternatives.

https://typescript-eslint.io/contributing/discussions

**Reply by baywet** [reply]
The reason why I posted here is because I wasn't sure if it was possible today. Hence I thought it could have been a feature suggestion.
